frontend/public/components/pod.tsx

```diff
@@ -571,7 +572,49 @@ const PodGraphs = requirePrometheus(({ pod }) => {
   );
 });
 
-export const PodStatus: React.FC<PodStatusProps> = ({ pod }) => <Status status={podPhase(pod)} />;
+const PodStatusPopover: React.FC<PodStatusPopoverProps> = ({
+  bodyContent,
+  headerContent,
+  status,
+}) => {
+  return (
+    <Popover headerContent={headerContent} bodyContent={bodyContent}>
+      <Button variant="link" isInline>
+        <Status status={status} />
+      </Button>
+    </Popover>
+  );
+};
+
+export const PodStatus: React.FC<PodStatusProps> = ({ pod }) => {
+  const status = podPhase(pod);
+  const unschedulableCondition = pod.status?.conditions?.find(
+    (condition) => condition.reason === 'Unschedulable' && condition.status === 'False',
+  );
+  const containerStatusStateWaiting = pod.status?.containerStatuses?.find(
+    (cs) => cs.state?.waiting,
+  );
+
+  if (status === 'Pending' && unschedulableCondition) {
+    return (
+      <PodStatusPopover
+        bodyContent={unschedulableCondition.message}
+        headerContent={`${PodModel.label} unschedulable`}
+        status={status}
+      />
+    );
+  }
+  if ((status === 'ErrImagePull' || status === 'ImagePullBackOff') && containerStatusStateWaiting) {
```

Comment:
I'm still not sure we need to check `status` at all. Should we simply show any `containerStatusStateWaiting` message if present?

Reply:
In testing, it looks like we end up with a popover with an empty message for `ContainerCreating` if we do that.  Not sure what other statuses may be affected...

Reply:
And we end up with a popover for `Terminating` as well.  Not sure that is desired.
<img width="479" alt="Screen Shot 2020-12-08 at 11 00 08 AM" src="https://user-images.githubusercontent.com/895728/101507510-a0025980-3944-11eb-90cc-9b183966ca94.png">

Reply:
We *do* want `CrashLoopBackoff` though.  My initial PR did not cover that.

Reply:
Updated to include `CrashLoopBackoff` and externalize `unschedulable`.